### PR TITLE
Add CIS Oracle Cloud Infrastructure Foundations Benchmark v1.2.0. Closes #61

### LINKED ISSUE #61: Add CIS Oracle Cloud Infrastructure Foundations Benchmark v1.2.0 - (04-14-2022)
**Is your feature request related to a problem? Please describe.**

CIS [reference](https://www.cisecurity.org/benchmark/oracle_cloud)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: cis_v120/section_2.sp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  tags = merge(local.cis_v120_2_common_tags, {
+    cis_item_id = "2.5"
+    cis_level   = "1"
+    cis_type    = "manual"
+    service     = "OCI/VCN"
+  })
+}
+
+control "cis_v120_2_6" {
```
Comment (rajlearner17) [inline cis_v120/section_2.sp:94]: @Priyanka-Chatterjee-2000 this is not added to cis_v120_2

File: cis_v120/section_2.sp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  tags = merge(local.cis_v120_2_common_tags, {
+    cis_item_id = "2.6"
+    cis_level   = "1"
+    cis_type    = "manual"
+    service     = "OCI/VCN"
+  })
+}
+
+control "cis_v120_2_7" {
```
Comment (rajlearner17) [inline cis_v120/section_2.sp:108]: Not added to cis_v120_2 benchmark, any specific reason?

File: cis_v120/section_2.sp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+  tags = merge(local.cis_v120_2_common_tags, {
+    cis_item_id = "2.7"
+    cis_level   = "1"
+    cis_type    = "manual"
+    service     = "OCI/VCN"
+  })
+}
+
+control "cis_v120_2_8" {
```
Comment (rajlearner17) [inline cis_v120/section_2.sp:122]: Not added to cis_v120_2 benchmark

File: cis_v120/section_2.sp [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+control "cis_v120_2_5" {
+  title         = "2.5 Ensure the default security list of every VCN restricts all traffic except ICMP"
+  description   = "A default security list is created when a Virtual Cloud Network (VCN) is created. Security lists provide stateful filtering of ingress and egress network traffic to OCI resources. It is recommended no security list allows unrestricted ingress access to Secure Shell (SSH) via port 22."
+  query         = query.core_default_security_list_allow_icmp_only
+  documentation = file("./cis_v120/docs/cis_v120_2_5.md")
+
+  tags = merge(local.cis_v120_2_common_tags, {
+    cis_item_id = "2.5"
+    cis_level   = "1"
+    cis_type    = "manual"
```
Comment (khushboo9024) [inline cis_v120/section_2.sp:92]:
```suggestion
    cis_type    = "automated"
```

File: cis_v120/section_3.sp [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    control.cis_v120_3_8,
+    control.cis_v120_3_9,
+    control.cis_v120_3_10,
+    control.cis_v120_3_11,
+    control.cis_v120_3_12,
+    control.cis_v120_3_13,
+    control.cis_v120_3_14,
+    control.cis_v120_3_15,
+    control.cis_v120_3_16,
+    control.cis_v120_3_17,
```
Comment (khushboo9024) [inline cis_v120/section_3.sp:27]:
```suggestion
    control.cis_v120_3_17
```

File: query/identity.sp [hunk 6]
```diff
@@ -71,6 +71,44 @@ query "identity_authentication_password_policy_strong_min_length_14" {
   EOQ
 }
 
+query "identity_allow_group_administrators_manage_all_resources_in_tenancy" {
+  sql = <<-EOQ
+    select
```
Comment (khushboo9024) [inline query/identity.sp:76]: This query should show result on the tenant level, If any group except administrators has got `manage all resources` permission then it should go in alarm .

File: cis_v120/cis.sp [hunk 7]
```diff
@@ -0,0 +1,23 @@
+locals {
+  cis_v120_common_tags = merge(local.oci_compliance_common_tags, {
+    cis         = "true"
+    cis_version = "v1.2.0"
+  })
+}
+
+benchmark "cis_v120" {
+  title         = "CIS v1.2.0"
+  description   = "The CIS Oracle Cloud Infrastructure Foundations Benchmark, provides prescriptive guidance for establishing a secure baseline configuration for the Oracle Cloud Infrastructure environment."
```
Comment (madhushreeray30) [inline cis_v120/cis.sp:10]:
```suggestion
  description   = "The CIS Oracle Cloud Infrastructure Foundations Benchmark provides prescriptive guidance for establishing a secure baseline configuration for the Oracle Cloud Infrastructure environment."
```

File: cis_v120/cis.sp [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  documentation = file("./cis_v120/docs/cis_overview.md")
+  children = [
+    benchmark.cis_v120_1,
+    benchmark.cis_v120_2,
+    benchmark.cis_v120_3,
+    benchmark.cis_v120_4,
+    benchmark.cis_v120_5,
```
Comment (madhushreeray30) [inline cis_v120/cis.sp:17]:
```suggestion
    benchmark.cis_v120_5
```

File: cis_v120/docs/cis_v120_1_4.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+Password policies are used to enforce password complexity requirements. IAM password policies can be used to ensure password are at least a certain length and are composed of certain characters. It is recommended the password policy require a minimum password length 14 characters and contain 1 non-alphabetic character (Number or “Special Character”).
+
+## Remediation
+
+### From Console
+
+**OCI Native IAM**
+
+1. Login to [OCI Console](https://www.oracle.com/cloud/)
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_1_4.md:11]:
```suggestion
1. Login to [OCI Console](https://www.oracle.com/cloud/).
```

File: cis_v120/docs/cis_v120_1_4.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+2. Go to `Identity` in the Services menu.
+3. Select `Authentication Settings` from the Identity menu.
+4. Click `Edit` in the middle of the page.
+5. Type the number `14` into the box below the text: `MINIMUM PASSWORD LENGTH (IN CHARACTERS)`.
+6. Select checkbox next to `MUST CONTAIN AT LEAST 1 SPECIAL CHARACTER` OR `MUST CONTAIN AT LEAST 1 NUMERIC CHARACTER`
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_1_4.md:16]:
```suggestion
6. Select checkbox next to `MUST CONTAIN AT LEAST 1 SPECIAL CHARACTER` OR `MUST CONTAIN AT LEAST 1 NUMERIC CHARACTER`.
```

File: cis_v120/docs/cis_v120_2_1.md [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+```bash
+oci network security-list get --security-list-id <security list id>
+```
+
+4. Then either:
+
+- Update the `security list`, copy the `ingress-security-rules` element from the JSON returned by the above get call, edit it appropriately and use it in the following command
+
+```bash
+oci network security-list update --security-list-id <security-list-id> -- ingress-security-rules '<ingress security rules JSON>'
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_2_1.md:48]:
```suggestion
oci network security-list update --security-list-id <security-list-id> --ingress-security-rules '<ingress security rules JSON>'
```

File: cis_v120/docs/cis_v120_2_6.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+5. Enter the following query in the query box:
+
+```bash
+query integrationinstance resources
+```
+
+6. For each OIC Instance returned click on the link under `Display name`
+7. Click on `Network Access`
+8 .Ensure `Restrict Network Access` is selected and the IP Address/CIDR Block
+as well as Virtual Cloud Networks are correct
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_2_6.md:20]:
```suggestion
8 .Ensure `Restrict Network Access` is selected and the IP Address/CIDR Block as well as Virtual Cloud Networks are correct
```

File: cis_v120/docs/cis_v120_2_6.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+11. Either edit the `Network Access` to be more restrictive
+
+### From Command Line
+
+1. Execute the following command:
+
+```bash
+for region in `oci iam region list | jq -r '.data[] | .name'`; 
+  do
+  for compid in `oci iam compartment list --compartment-id-in-subtree TRUE 2>/dev/null | jq -r '.data[] | .id'`
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_2_6.md:33]:
```suggestion
   for compid in `oci iam compartment list --compartment-id-in-subtree TRUE 2>/dev/null | jq -r '.data[] | .id'`
```

File: cis_v120/docs/cis_v120_2_7.md [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+5. Enter the following query in the query box:
+
+```bash
+query analyticsinstance resources
+```
+
+6. For each OAC Instance returned click on the link under `Display name`.
+7. Ensure `Access Control Rules` IP Address/CIDR Block as well as Virtual Cloud Networks are correct.
+8. Repeat for other subscribed regions.
+9.  For each OAC instance in the returned results, click the OAC Instance name
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_2_7.md:20]:
```suggestion
9. For each OAC instance in the returned results, click the OAC Instance name
```

File: cis_v120/docs/cis_v120_2_7.md [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    if [ ! -z "$output" ]; then echo $output; fi
+    done 
+  done
+```
+
+2. Ensure `network-endpoint-type` are correct.
+3. Get the json input format by executing the below command:
+
+```bash
+oci analytics analytics-instance change-network-endpoint --generate-full- command-json-input
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_2_7.md:44]:
```suggestion
oci analytics analytics-instance change-network-endpoint --generate-full-command-json-input
```

File: cis_v120/docs/cis_v120_2_8.md [hunk 16]
```diff
@@ -0,0 +1,59 @@
...
+4. Click the `Advanced Resource Query` button in the upper right of the screen.
+5. Enter the following query in the query box:
+
+```bash
+query autonomousdatabase resources
+```
+
+6. For each ABD-S database returned click on the link under `Display name`
+7. Click `Edit` next to `Access Control List`
+8. Ensure `Access Control Rules’ IP Address/CIDR Block as well as VCNs are correct
```
[the remark this answers is not in the document]
Comment (madhushreeray30) [reply cis_v120/docs/cis_v120_2_8.md:20]: It is wrong in the document but if we don't change following points will render incorrectly

File: cis_v120/docs/cis_v120_3_2.md [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+## Remediation
+
+### From Console
+
+1. Go to the [Tag Namespaces page](https://console.us-ashburn1.oraclecloud.com/identity/tag-namespaces)
+2. Select the Root `compartment`
+3. Click `Create Namespace Definition`
+4. Create the `Namespace` in the `Root Compartment`
+5. Click on the newly created `Namespace definition`
+6. Click `Create Tag Key Definition` Create a Tag key definition by providing a `Tag Key`, `Description` and selecting `“Static Value”` for `Tag Value Type`
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_3_2.md:14]:
```suggestion
6. Click `Create Tag Key Definition` 
Create a Tag key definition by providing a `Tag Key`, `Description` and selecting `“Static Value”` for `Tag Value Type`
```

File: cis_v120/docs/cis_v120_4_2_2.md [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+query bootvolume resources
+```
+
+6. For each boot volume returned click on the link under `Display name`
+7. Ensure `Encryption Key` does not say `Oracle managed key`
+8. Repeat for other subscribed regions
+9. For each Boot Volume in the returned results, click the Boot Volume name
+10. Click `Assign` next to `Encryption Key`
+11. Select the `Vault Compartment` and `Vault`
+12.  Select the `Master Encryption Key Compartment` and `Master Encryption key`
```
Comment (madhushreeray30) [inline cis_v120/docs/cis_v120_4_2_2.md:25]:
```suggestion
12. Select the `Master Encryption Key Compartment` and `Master Encryption key`
```

File: cis_v120/section_4.sp [hunk 19]
```diff
@@ -0,0 +1,158 @@
...
+  ]
+
+  tags = merge(local.cis_v120_4_common_tags, {
+    service = "OCI/Storage"
+    type    = "Benchmark"
+  })
+}
+
+benchmark "cis_v120_4_1" {
+  title         = "4.1 Object Storage"
```
Comment (misraved) [inline cis_v120/section_4.sp:35]: Is the title correct? I see a similar title being used for `cis_v120_4` benchmark.

**Review by misraved — COMMENTED** [review]
@Priyanka-Chatterjee-2000 please take a look at the review comments. Thanks!!

Also could you please update the docs file with the latest CIS benchmark references?

File: query/storage.sp [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+      left join oci_identity_compartment as c on c.id = a.compartment_id;
+  EOQ
+}
+
+query "objectstorage_bucket_cmk_encryption_enabled" {
+  sql = <<-EOQ
+    select
+      a.id as resource,
+      case
+        when kms_key_id is not null and kms_key_id <> '' then 'ok'
```
[the remark this answers is not in the document]
Comment (Priyanka-Chatterjee-2000) [reply query/storage.sp:48]: We took the reference from our previous pattern to add these queries: https://hub.steampipe.io/mods/turbot/oci_insights/queries/objectstorage_bucket_encryption, https://hub.steampipe.io/mods/turbot/oci_insights/queries/blockstorage_block_volume_encryption, https://hub.steampipe.io/mods/turbot/oci_insights/queries/blockstorage_boot_volume_encryption, https://hub.steampipe.io/mods/turbot/oci_insights/queries/filestorage_filesystem_encryption

**Review by misraved — APPROVED** [review]

**Pull request merged**